### PR TITLE
Fix: Add support for filtering challenges for hosts.

### LINKED ISSUE #1141: Add support for filtering challenges on the basis of challenge phase for host.
**Current Behaviour**: The filtering of submissions under the `View All Submissions` tab for a challenge host is on the basis of `challenge_pk` only. The results are same even if challenge host selects any `challenge_phase`.
**Expected Behaviour**: We want to filter the submissions on the basis of the both `challenge_pk` and `challenge_phase` corresponding to that challenge, so that the result set will only contain the submissions corresponding to the `challenge_phase` that the challenge host selects.

Please let me know in case of further queries.

### PR DESCRIPTION
**Pull request by isht3**
Fixes #1141

### CODE REVIEW

File: apps/challenges/views.py [hunk 1]
```diff
@@ -707,8 +707,7 @@ def get_all_submissions_of_challenge(request, challenge_pk, challenge_phase_pk):
 
         # Filter submissions on the basis of challenge for host for now. Later on, the support for query
         # parameters like challenge phase, date is to be added.
-
-        submissions = Submission.objects.filter(challenge_phase__challenge=challenge).order_by('-submitted_at')
+        submissions = Submission.objects.filter(challenge_phase=challenge_phase).order_by('-submitted_at')
```
Comment (RishabhJain2018) [inline apps/challenges/views.py]: @isht3 Please add support for filtering on the basis of `challenge_phase` corresponding to a `challenge`.

**Comment by RishabhJain2018** [comment]
@isht3 Moreover please follow these guidelines https://help.github.com/articles/closing-issues-using-keywords/ for mentioning a issue in a pull request.

File: apps/challenges/views.py (hunk 1 above)
Comment (isht3) [reply apps/challenges/views.py]: @RishabhJain2018 On line 700 `challenge_phase = ChallengePhase.objects.get(pk=challenge_phase_pk, challenge=challenge)` and I used this `challenge_phase` to obtain its submissions. This `challenge_phase` is of that particular challenge, right?

Comment (RishabhJain2018) [reply apps/challenges/views.py]: @isht3 Yes you are right! My bad, I didn't see that line during review.

Comment (isht3) [reply apps/challenges/views.py]: @RishabhJain2018 Totally cool! Ill start with the other issue. :smile:

**Comment by RishabhJain2018** [comment]
@isht3 Can you please check why travis build is failing.

**Comment by isht3** [comment]
@RishabhJain2018 Done!

**Comment by RishabhJain2018** [comment]
@isht3 Please resolve the merge conflicts.

**Comment by RishabhJain2018** [comment]
@isht3 Can you please check why travis build is failing.

**Comment by isht3** [comment]
@RishabhJain2018 Done!

**Pull request merged**